### ISSUE TITLE
New sudo.conf makes statically compiled sudo unusable

### ISSUE THREAD

**Issue by michael-o** [sudo-project/sudo#61]
Compiled from master on HP-UX with aCC:
```
$CONFIGURE --sysconfdir=$SYSCONFDIR --disable-root-sudo --disable-sasl --disable-pam-session \
  --with-env-editor --without-ldap --disable-shared-libutil --enable-static-sudoers \
  --disable-log-server --disable-log-client --enable-gss-krb5-ccache-name
```
Note: `--disable-shared-libutil --enable-static-sudoers`.

`/opt/ports/etc/sudo.conf` contains:
```
Plugin sudoers_policy sudoers.so
Plugin sudoers_io sudoers.so
Plugin sudoers_audit sudoers.so
```

but `sudoers.so` is not present:
```
osipovmi@deblndw024v:/var/tmp/ports/work/sudo (master *%=)
$ find /opt/ports -name sudoers.so
osipovmi@deblndw024v:/var/tmp/ports/work/sudo (master *%=)
$ ll /opt/ports/libexec/sudo/
total 3310
-r-xr-xr-x 1 root root   1026 2020-09-03 10:36 audit_json.la*
-r-xr-xr-x 1 root root 828124 2020-09-03 10:36 audit_json.so*
-r-xr-xr-x 1 root root   1026 2020-09-03 10:36 group_file.la*
-r-xr-xr-x 1 root root 827544 2020-09-03 10:36 group_file.so*
-r-xr-xr-x 1 root root   1056 2020-09-03 10:36 sample_approval.la*
-r-xr-xr-x 1 root root 823920 2020-09-03 10:36 sample_approval.so*
-r-xr-xr-x 1 root root    954 2020-09-03 10:36 sudo_noexec.la*
-r-xr-xr-x 1 root root  74984 2020-09-03 10:36 sudo_noexec.so*
-r-xr-xr-x 1 root root   1038 2020-09-03 10:36 system_group.la*
-r-xr-xr-x 1 root root 822288 2020-09-03 10:36 system_group.so*

```
and when I start sudo it says:
```
$ LANG=C sudo
sudo: error in /opt/ports/etc/sudo.conf, line 16 while loading plugin "sudoers_audit"
sudo: unable to find symbol "sudoers_audit" in sudoers.so
sudo: fatal error, unable to load plugins
```
as soon as I remove `Plugin sudoers_audit sudoers.so` sudo does work.

Did I do anything or is this a config glitch?

**Comment by millert** [comment]
Fixed by 24b35393e3c38c483efb10c8e45a3ba0d1522bf5

**Comment by michael-o** [comment]
Brilliant, thank you!

**Comment by GuillaumeDua** [comment]
_My apologizes for digging this issue up two years later._

Naively, should `sudoers_audit` be a value of type `audit_plugin` ?
https://github.com/sudo-project/sudo/commit/24b35393e3c38c483efb10c8e45a3ba0d1522bf5#diff-3ebe283c4fb7df34e58785f0373eb05ec83c5664abed4d85d0702dbfd942cc48R45

At best this is an inconsistency, at worst something error-prone as the two types are not isomorphic (11 vs. 15 fields).

**Comment by millert** [comment]
Yes, you are correct, that was a pasto on my part.  It is basically harmless since we just store a pointer to the struct and it gets cast to the correct type when used.  I have fixed it in [4d58fc9](https://github.com/sudo-project/sudo/commit/4d58fc904da6f8df7f1699b0c0d5d50060e039dd).
